Modules/posixmodule.c

```diff
@@ -12916,14 +12916,15 @@ _pystatvfs_fromstructstatfs(PyObject *module, struct statfs st) {
 
     _Static_assert(sizeof(st.f_blocks) == sizeof(long long), "assuming large file");
 
-#define SET_ITEM(v, index, item)                   \
-    do {                                           \
-        if (item == NULL) {                        \
-            Py_DECREF(v);                          \
-            return NULL;                           \
-        }                                          \
-        PyStructSequence_SET_ITEM(v, index, item); \
-    } while (0)                                    \
+#define SET_ITEM(v, index, expr)                    \
+    do {                                            \
+        PyObject *obj = (expr);                     \
+        if (obj == NULL || PyErr_Occurred()) {      \
+            Py_DECREF(v);                           \
+            return NULL;                            \
+        }                                           \
+        PyStructSequence_SET_ITEM(v, (index), obj); \
```

Comment:
```suggestion
        PyStructSequence_SET_ITEM((v), (index), obj); \
```
Parenthesize all macro arguments for paranoia reasons